### PR TITLE
Implement card Attack Pattern Delta (SOR 106)

### PR DESCRIPTION
**Pull request by garrettdc2**
SWUDB: https://swudb.com/card/SOR/106/attack-pattern-delta

image

### CODE REVIEW

File: server/game/cards/01_SOR/events/AttackPatternDelta.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    private buildModifyStatsForPhaseImmediateEffect(modificationAmount: number) {
+        return AbilityHelper.immediateEffects.forThisPhaseCardEffect({
+            effect: AbilityHelper.ongoingEffects.modifyStats({ power: modificationAmount, hp: modificationAmount }),
+        });
+    }
+
+    public override setupCardAbilities () {
+        this.setEventAbility({
+            title: 'Give a friendly unit +3/+3 for this phase.',
+            targetResolver: {
```
Comment (AMMayberry1) [inline server/game/cards/01_SOR/events/AttackPatternDelta.ts:28]: These need to be checking that it's a friendly unit (and the tests updated accordingly). Should just be `controller: RelativePlayer.Self`

File: test/server/cards/01_SOR/events/AttackPatternDelta.spec.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                    ]);
+
+                    // Case 6: The player selects Red Three. It gets +1/+1.
+                    context.player1.clickCard(context.redThree);
+
+                    expect(context.redThree.getPower()).toEqual(3);
+                    expect(context.redThree.getHp()).toEqual(4);
+
+                    // Case 7: The buffs wear off at the end of the phase.
+                    context.player2.passAction();
```
Comment (AMMayberry1) [inline test/server/cards/01_SOR/events/AttackPatternDelta.spec.ts:65]: Can use `context.moveToNextActionPhase()`

File: test/server/cards/01_SOR/events/AttackPatternDelta.spec.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                        groundArena: ['battlefield-marine', 'wampa'],
+                        spaceArena: ['red-three#unstoppable']
+                    },
+                    player2: {
+                        leader: { card: 'luke-skywalker#faithful-friend', deployed: true },
+                    }
+                });
+            });
+
+            it(
```
Comment (AMMayberry1) [inline test/server/cards/01_SOR/events/AttackPatternDelta.spec.ts:18]: nit: the test name gets concatenated together with the describe name to form a sentence, so typically the test case name starts with lowercase "should". If you want to inspect the result you can just force the test to fail and you'll see the name

File: server/game/cards/01_SOR/events/AttackPatternDelta.ts (hunk 1 above)
Comment (garrettdc2) [reply server/game/cards/01_SOR/events/AttackPatternDelta.ts:28]: Done!

File: test/server/cards/01_SOR/events/AttackPatternDelta.spec.ts (hunk 3 above)
Comment (garrettdc2) [reply test/server/cards/01_SOR/events/AttackPatternDelta.spec.ts:18]: Fixed.

File: test/server/cards/01_SOR/events/AttackPatternDelta.spec.ts (hunk 2 above)
Comment (garrettdc2) [reply test/server/cards/01_SOR/events/AttackPatternDelta.spec.ts]: Done

File: test/server/cards/01_SOR/events/AttackPatternDelta.spec.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                    // Case 7: The buffs wear off at the end of the phase.
+                    context.moveToNextActionPhase();
+
+                    expect(context.wampa.getPower()).toEqual(4);
+                    expect(context.wampa.getHp()).toEqual(5);
+                    expect(context.battlefieldMarine.getPower()).toEqual(3);
+                    expect(context.battlefieldMarine.getHp()).toEqual(3);
+                    expect(context.redThree.getPower()).toEqual(2);
+                    expect(context.redThree.getHp()).toEqual(3);
+                }
```
Comment (Ennis0) [inline test/server/cards/01_SOR/events/AttackPatternDelta.spec.ts:73]: I'd add tests for when the controller has only one and two units, making sure the extraneous buffs don't get applied.

Comment (garrettdc2) [reply test/server/cards/01_SOR/events/AttackPatternDelta.spec.ts:73]: I can do that 👍

Comment (garrettdc2) [reply test/server/cards/01_SOR/events/AttackPatternDelta.spec.ts:71]: Done.

**Review by AMMayberry1 — APPROVED** [review]

**Pull request merged**